packages/generator-joplin/tools/updateCategories.ts

```diff
@@ -0,0 +1,27 @@
+
+import { dirname } from 'path';
+import { writeFile, readFile } from 'fs/promises';
+const allCategories = require('@joplin/lib/pluginCategories.json');
+
+const updateCategories = async () => {
+	const buildScriptPath = `${dirname(__dirname)}/generators/app/templates/webpack.config.js`;
+	const originalBuildScript = await readFile(buildScriptPath, 'utf8');
+
+	const autoGeneratedStartString = '\n// Begin auto-generated';
+	const autoGeneratedStart = originalBuildScript.indexOf(autoGeneratedStartString);
+	const autoGeneratedEndString = '// End auto-generated\n';
+	const autoGeneratedEnd = originalBuildScript.indexOf(autoGeneratedEndString);
+
+	// Make the generated categories pass the linter
+	const possibleCategoriesText = JSON.stringify(allCategories).replace(/["]/g, '\'');
```

Comment:
We have a function called `insertContentIntoFile` to do this kind of operation, is it possible to use it here?

Also in general I add to the comment where the code was generated from, so that it's easier to find it back and change it if needed, for example `// AUTO-GENERATED by generate-database-type`